compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -2305,6 +2304,43 @@ impl<'a> Parser<'a> {
                             pat.span.shrink_to_lo(),
                         )
                     }
+                    PatKind::Ref(ref inner_pat, _) => {
```

Comment:
I don't like that the "explicitly ignore the parameter name" code is duplicated from below. Could we fix this by making this line:
```rust
PatKind::Ref(ref inner_pat, _) if let PatKind::Ref(inner_inner_pat, _) = &inner_pat.kind => {
```
Then the `0|1` arm should be unreachable. Please explicitly mark it as `unreachable!()` so our assumptions are clear. Feel free to make the names of my variables a bit better :P